packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -45,8 +63,76 @@ export async function createReceiverService(
   }
 
   return {
-    get: (url) => getReceiver(deps, url)
+    get: (url) => getReceiver(deps, url),
+    create: (url) => createReceiver(deps, url)
+  }
+}
+
+async function createReceiver(
+  deps: ServiceDependencies,
+  args: CreateReceiverArgs
+): Promise<Receiver | ReceiverError> {
+  const localPaymentPointer = await deps.paymentPointerService.getByUrl(
+    args.paymentPointerUrl
+  )
+
+  const incomingPaymentOrError = localPaymentPointer
+    ? await createLocalIncomingPayment(deps, args, localPaymentPointer)
+    : await deps.remoteIncomingPaymentService.create(args)
+
+  if (isReceiverError(incomingPaymentOrError)) {
+    return incomingPaymentOrError
   }
+
+  const receiver = Receiver.fromIncomingPayment(incomingPaymentOrError)
+
+  if (!receiver) {
+    const errorMessage = 'Could not create receiver from incoming payment'
+    deps.logger.error({ incomingPayment: incomingPaymentOrError }, errorMessage)
+
+    throw new Error(errorMessage)
+  }
+
+  return receiver
+}
+
+async function createLocalIncomingPayment(
+  deps: ServiceDependencies,
+  args: CreateReceiverArgs,
+  paymentPointer: PaymentPointer
+): Promise<OpenPaymentsIncomingPayment | ReceiverError> {
+  const { description, expiresAt, incomingAmount, externalRef } = args
+
+  const incomingPaymentOrError = await deps.incomingPaymentService.create({
+    paymentPointerId: paymentPointer.id,
+    description,
+    expiresAt,
+    incomingAmount,
+    externalRef
+  })
+
+  if (isIncomingPaymentError(incomingPaymentOrError)) {
+    const errorMessage = 'Could not create local incoming payment'
+    deps.logger.error(
+      { error: receiverErrorToMessage(incomingPaymentOrError) },
+      errorMessage
+    )
+
+    return incomingPaymentOrError
+  }
+
+  const connection = deps.connectionService.get(incomingPaymentOrError)
+
+  if (!connection) {
+    const errorMessage = 'Could not get connection for local incoming payment'
+    deps.logger.error({ incomingPaymentOrError }, errorMessage)
+
+    throw new Error(errorMessage)
```

Comment:
Maybe I can also return an additional type of `ReceiverError` here as well, together with L93?